notebooks-auth/resources/user_test_cowbird_jupyter.ipynb

```diff
@@ -0,0 +1,172 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #9.        for file, expected_errno, expected_write in [(public_wps_outputs_filepath, 30, False), 
</pre><p>It would be useful to have a URL pointing at the definition where those permissions get defined such that the expected access can be easily understood.</p><p> </p><p>Also, for quick static analysis, indicate what OS errors 1 and 30 represent respectively.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Ouranosinc/PAVICS-e2e-workflow-tests/pull/131/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Reply:
<p>For the URLs reference, do you mean where the permissions are assigned in the actual code of birdhouse or Cowbird, or maybe the Cowbird's documentation?</p>

Reply:
I think mostly the birdhouse-deploy component where they are defined would be relevant. Other references could be inferred from there after.

Reply:
Well, the permissions are really assigned mostly in Cowbird :
- shapefile datastore : permissions are set on the directory with our latest update on Cowbird
- user WPS outputs : handled entirely by Cowbird, which creates the hardlinks on his side
-  public WPS outputs : Only one handled by birdhouse, from the jupyterhub config, where the volume is mounted as read-only

And for the chown/chmod part, it's supposed to be handled by Cowbird too.

That's why I was asking, because it's a bit spread out. Either I would put the links to the different places in the codes those permissions are set, or maybe just link the Cowbird doc, either I wouldn't put anything. 
The permissions here are not really related to the optional-component.

Reply:
I was referring mostly to https://github.com/bird-house/birdhouse-deploy/blob/cowbird-jupyter-e2e-test/birdhouse/optional-components/test-cowbird-jupyter-access/test_cowbird_jupyter.py that pre-configures a lot of items with `TEST_USER`. 

From then on, it is easy to infer that Cowbird config, its logic, and other services which are involved as well.